src/L2VoteAggregator.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import {ERC20Votes} from "openzeppelin/token/ERC20/extensions/ERC20Votes.sol";
+import {SafeCast} from "openzeppelin/utils/math/SafeCast.sol";
+import {IWormhole} from "wormhole/interfaces/IWormhole.sol";
+
+import {L2GovernorMetadata} from "src/L2GovernorMetadata.sol";
+import {IL1Block} from "src/interfaces/IL1Block.sol";
+
+/// @notice A contract to collect votes on L2 to be bridged to L1.
+contract L2VoteAggregator {
+  /// @notice The number of blocks before L2 voting closes. We close voting 1200 blocks before the
+  /// end of the proposal to cast the vote. We will allow another 1200 blocks to allow for the final
+  /// vote total to be sent to the L1.
+  uint32 public constant CAST_VOTE_WINDOW = 2400;
+
+  /// @notice The Wormhole contract to bridge messages to L1.
+  IWormhole immutable CORE_BRIDGE;
+
+  /// @notice A unique number used to send messages.
+  uint32 public nonce;
+
+  /// @notice The token used to vote on proposals provided by the `GovernorMetadata`.
+  ERC20Votes immutable VOTING_TOKEN;
+
+  /// @notice The `GovernorMetadata` contract that provides proposal information.
+  L2GovernorMetadata immutable GOVERNOR_METADATA;
+
+  /// @notice The contract that handles fetch the L1 block on the L2.
+  IL1Block immutable L1_BLOCK;
+
+  /// @dev Thrown when an address has no voting weight on a proposal.
+  error NoWeight();
+
+  /// @dev Thrown when an address has already voted.
+  error AlreadyVoted();
+
+  /// @dev Thrown when an invalid vote is cast.
+  error InvalidVoteType();
+
+  /// @dev Thrown when proposal is inactive.
+  error ProposalInactive();
+
+  /// @dev The voting options corresponding to those used in the Governor.
+  enum VoteType {
+    Against,
+    For,
+    Abstain
+  }
+
+  /// @dev Data structure to store vote preferences expressed by depositors.
+  // TODO: Does it matter if we use a uint128 vs a uint256?
```

Comment:
Highlighting this question

Reply:
I don't think it should, no

Reply:
Should I make an issue than to change to uint256?

Reply:
Nope I think it's fine as uint128